src/main/scala/pt/tecnico/dsi/neutron/models/Network.scala

```diff
@@ -1,58 +1,62 @@
 package pt.tecnico.dsi.neutron.models
 
-import java.time.LocalDateTime
-
 import io.circe.derivation.{deriveDecoder, deriveEncoder, renaming}
-import io.circe.{Decoder, Encoder, HCursor}
+import io.circe.{Decoder, Encoder}
 
 object Network {
 
-  // TODO: Rethink this
   object Read {
-    implicit val decoder: Decoder[Read] = (c: HCursor) => {
-      for {
-        providerNetworkType <- c.downField("provider:network_type").as[String]
-        providerPhysicalNetwork <- c.downField("provider:physical_network").as[String]
-        providerSegmentationId <- c.downField("provider:segmentation_id").as[Integer]
-        routerExternal <- c.downField("router:external").as[Boolean]
-        network <- c.as[Read](deriveDecoder(renaming.snakeCase))
-      } yield network.copy(
-        providerNetworkType = providerNetworkType,
-        providerPhysicalNetwork = providerPhysicalNetwork,
-        providerSegmentationId = providerSegmentationId,
-        routerExternal = routerExternal
-      )
+
+    def decoderAfterRename[T](m: Map[String, String], d: Decoder[T]): Decoder[T] = {
+      d.prepare {
+        _.withFocus {
+          _.mapObject { x =>
+            m.foldLeft(x) { (a, b) =>
+              val value = x(b._1)
+              a.add(b._2, value.get).remove(b._1)
+            }
+          }
+        }
+      }
     }
+
+    implicit val decoder: Decoder[Read] = decoderAfterRename[Read](
+      Map(
+        "provider:network_type" -> "provider_network_type",
+        "provider:physical_network" -> "provider_physical_network",
+        "provider:segmentation_id" -> "provider_segmentation_id",
+        "router:external" -> "router_external",
+      ), deriveDecoder(renaming.snakeCase))
   }
 
   sealed case class Read(
     adminStateUp: Boolean,
     availabilityZoneHints: List[String], // ???
     availabilityZones: List[String], // ???
-    createdAt: LocalDateTime,
+    createdAt: String,
```

Comment:
This should be a DateTime, probably a ZonedDateTime.

Reply:
ZonedDateTimes don't seem to look like `2020-07-08T20:06:16Z`, at least in the [docs](https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html).

I'll check it out when I get home.